nni/common/concrete_trace_utils/operator_patcher.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from .concrete_tracer import ConcreteTracer
+
+import ast
+import inspect
+import logging
+
+from textwrap import dedent
+from types import MethodType, FunctionType
+from typing import List, Optional, Callable, Dict
+
+import torch
+
+from .utils import (
+    _orig_type,
+    _orig_isinstance,
+    _orig_len,
+    _orig_dict,
+    _orig_zip,
+)
+
+_logger = logging.getLogger(__name__)
+
+class TransformerOp(ast.NodeTransformer):
```

Comment:
"Transformer" in "TransformerOp" is a little misleading, here it means "transformation" rather than "transformer" models, right?

Reply:
yes. it's a 'python func transformer' not something occurred in training process.